hawk/app/models/crm_config.rb

```diff
@@ -48,60 +48,71 @@ def persisted?
   end
 
   class << self
+    def get_parameters_from(crm_config, cmd)
+      #todo: this doesn't work with safe_x. research why.
+      REXML::Document.new(%x[#{cmd} 2>/dev/null]).tap do |xml|
+        return unless xml.root
+
+        xml.elements.each("//parameter") do |param|
+          name = param.attributes["name"]
+          content = param.elements["content"]
+          shortdesc = param.elements["shortdesc[@lang=\"#{I18n.locale.to_s.gsub("-", "_")}\"]|shortdesc[@lang=\"en\"]"].text || ""
+          longdesc  = param.elements["longdesc[@lang=\"#{I18n.locale.to_s.gsub("-", "_")}\"]|longdesc[@lang=\"en\"]"].text || ""
+
+          type = content.attributes["type"]
+          default = content.attributes["default"]
+
+          advanced = param.attributes["advanced"] || shortdesc.match(/advanced use only/i) || longdesc.match(/advanced use only/i)
+
+          crm_config[name] = {
+            type: content.attributes["type"],
+            readonly: false,
+            shortdesc: shortdesc,
+            longdesc: longdesc,
+            advanced: (advanced and (advanced=="1")) ? true : false,
+            default: default
+          }
+
+          if type == "enum"
+            match = longdesc.match(/Allowed values:(.*)/i)
```

Comment:
And we should first try to retrieve the allowed values from the "option" elements under the "content" if appearing:

https://github.com/ClusterLabs/pacemaker/pull/3297/commits/5ef28e37d6f3008a55a7621262a5acd3be637c16#diff-14ca730a938353184deb85c9b9e22bc6e448635cfc7fbb62f8a80036ab97fe32R67-R70

Parsing from any "Allowed values ..." should be a fallback.

Reply:
Good point. I didn't change this block but simply moved it. I will add retrieving the allowed values from the option as sub-feature in the next PR.